src/v3/src/transformer/field/transform.ts

```diff
@@ -41,28 +49,72 @@ const mapUiElement = (input: Input): FieldElement => {
   };
 };
 
-const getValidationMessages = (
-  fieldName: string,
+const getValidationMessages: ValidationErrorTransformer = (
+  input: Input,
+  data: Record<string, unknown>,
   widgetProps: WidgetProps,
   step?: NextStep,
-): IdxMessage[] => {
+): WidgetMessage[] | undefined => {
+  const { name: fieldName, type, maxLength } = input;
   const { name } = step || {};
-  const errorMessage: IdxMessage = {
-    class: 'ERROR',
-    message: loc('model.validation.field.blank', 'login'),
-    i18n: { key: 'model.validation.field.blank' },
-  };
+  const fieldValue = data[fieldName];
+
+  // Customized i18n key support for fields
   const customizedErrorConfig = [
     { field: 'identifier', key: 'error.username.required' },
     { field: 'credentials.passcode', key: 'error.password.required' },
   ].find((obj) => obj.field === fieldName);
-  if (customizedErrorConfig
-    && name === IDX_STEP.IDENTIFY
-    && isCustomizedI18nKey(customizedErrorConfig.key, widgetProps)) {
-    errorMessage.message = loc(customizedErrorConfig.key, 'login');
-    errorMessage.i18n.key = customizedErrorConfig.key;
+  const useCustomizedBlankErrorMessage = !!customizedErrorConfig && name === IDX_STEP.IDENTIFY
+                                    && isCustomizedI18nKey(customizedErrorConfig.key, widgetProps);
+
+  const populatedFieldErrorChecks: ValidationErrorTester[] = [
+    {
+      // Validate maxLength for string inputs that have the property set
+      tester: (value?: unknown) => !!maxLength && type === 'string' && !!value && (value as string).length > maxLength,
```

Comment:
While we are here, we can probably also define a validator for min length (i.e in Enroll profile allows org admins to set the string length minimums and maximums). 

This is the key:
`model.validation.field.string.minLength`